applications/ParticleMechanicsApplication/custom_utilities/mpm_particle_generator_utility.h

```diff
@@ -50,11 +50,227 @@ namespace MPMParticleGeneratorUtility
+
+                    // Evaluation of element area/volume
+                    // TODO this should be corrected, in the general case the mass is not evenly distributed.
+                    const double area = r_geometry.Area();
+                    if (domain_size == 2 && i->GetProperties().Has(THICKNESS)) {
+                        const double thickness = i->GetProperties()[THICKNESS];
+                        mp_mass[0] = area * thickness * density / integration_point_per_elements;
+                    }
+                    else {
+                        mp_mass[0] = area * density / integration_point_per_elements;
+                    }
+                    mp_volume[0] = area / integration_point_per_elements;
+
+                    // Loop over the material points that fall in each grid element
+                    unsigned int new_element_id = 0;
+                    for (unsigned int PointNumber = 0; PointNumber < integration_point_per_elements; PointNumber++)
+                    {
+                        std::vector<double> MP_density = { density };
+
+                        xg[0].clear();
+
+                        // Loop over the nodes of the grid element
+                        for (unsigned int dimension = 0; dimension < r_geometry.WorkingSpaceDimension(); dimension++)
+                        {
+                            for (unsigned int j = 0; j < r_geometry.size(); j++)
+                            {
+                                xg[0][dimension] = xg[0][dimension] + shape_functions_values(PointNumber, j) * r_geometry[j].Coordinates()[dimension];
+                            }
+                        }
+                        if (IsAxisSymmetry)  mp_mass[0] = mp_volume[0] * 2.0 * Globals::Pi * xg[0][0] * density;
+
+                        typename BinBasedFastPointLocator<TDimension>::ResultIteratorType result_begin = results.begin();
+                        Element::Pointer pelem;
+                        Vector N;
+
+                        // FindPointOnMesh find the background element in which a given point falls and the relative shape functions
+                        bool is_found = SearchStructure.FindPointOnMesh(xg[0], N, pelem, result_begin);
+                        if (!is_found) KRATOS_WARNING("MPM particle generator utility") << "::search failed." << std::endl;
+
+                        pelem->Set(ACTIVE);
+                        auto p_new_geometry = CreateQuadraturePointsUtility<Node<3>>::CreateFromCoordinates(
+                            pelem->pGetGeometry(), xg[0],
+                            mp_volume[0]);
+
+                        // Create new material point element
+                        new_element_id = last_element_id + PointNumber;
+                        Element::Pointer p_element = new_element.Create(
+                            new_element_id, p_new_geometry, properties);
+
```

Comment:
short question: why are the nodes of the background mesh not set active here (similar to the search function)?

Reply:
The nodes aren't set active here because the search function is called at the start of every timestep (and they are set active there, as you say)